ci: improve maintenance of test containers

Henrik66 opened the pull request:
Install AUR helper (paru) into Arch test container and improve maintenance of test containers.

Pulls in https://github.com/dracutdevs/dracut/pull/2427 as one of the commits in this PR.

## Checklist
- [X] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #2422 #2433

Henrik66 committed:
ci: install AUR helper into Arch test container

Install more packages to enable more testing on Arch.

LaszloGombos reviewed (commented):
Thanks @Henrik66 . 

Can you please also add `pcsclite` package ? 

Can you see if you can find a way not to install `mkinitcpio` (the default `initramfs` provider on arch) ? There is a possibly solution used in the `Debian` container to avoid having `initramfs-tools` installed.

Henrik66 commented on test/container/Dockerfile-Arch:
```diff
@@ -2,16 +2,64 @@ FROM docker.io/archlinux
-RUN pacman -U --noconfirm ~build/tgt/*.pkg.tar.*
-RUN rm -fr ~build
+# Install needed packages for the dracut CI container
+RUN su build -c 'paru --noconfirm -Syu \
```
Ok. Done.

Henrik66 commented on test/container/Dockerfile-Arch:
```diff
@@ -2,16 +2,64 @@ FROM docker.io/archlinux
+    nbd \
+    ndctl \
+    networkmanager \
+    nfsidmap \
```
Ok. Done.

Henrik66 commented on test/container/Dockerfile-Arch:
```diff
@@ -2,16 +2,64 @@ FROM docker.io/archlinux
+    e2fsprogs \
+    f2fs-tools \
+    fuse3 \
+    glibc \
```
Ok. Done.

Henrik66 commented:
> Can you see if you can find a way not to install `mkinitcpio` (the default `initramfs` provider on arch) ? There is a possibly solution used in the `Debian` container to avoid having `initramfs-tools` installed.

I found out that [assume-installed](https://archlinux.org/pacman/pacman.8.html#:~:text=%2D%2D-,assume%2Dinstalled,-%3Cpackage%3Dversion%3E) option can do this.

LaszloGombos reviewed (approved):
Thanks

Henrik66 commented:
Start working on https://github.com/dracutdevs/dracut/issues/2433 .

Pulling in https://github.com/dracutdevs/dracut/pull/2427 as this PR would help with https://github.com/dracutdevs/dracut/issues/2433 .

Marking in draft for now as I will likely need help with this PR.

Henrik66 committed:
ci: improve maintenance of test containers

Consolidate list of package generation into test-packages.sh tool.

Add jq to all test containers.

Remove g++, less and quilt from Debian test container.

Add iscsiuio to dpkg based distributions.

LaszloGombos reviewed (approved):
LGTM. Hopefully this helps with maintenance - and at least now we have a list of common packages installed in the containers.

bdrung commented:
The test code is harder to read and harder to copy-paste after this change.

The list would be easier to maintain if the entries would be grouped by topic. Example question: Why is btrfs-progs not installed on openSUSE? There are many packages that just have different names. Having those in one place would help, but I have no idea how to codify that in a readable way.

```
opensuse-tumbleweed: ShellCheck, others: shellcheck
fedora|opensuse-tumbleweed: kernel, debian|ubuntu: linux-image-generic
```

bdrung commented:
> Some documentation also references package names - https://github.com/dracutdevs/dracut/blob/master/docs/README.generic . We should either remove these or replace them with some syntax that considers several distributions.

Is this list referring to projects or package names?

Also "Generic initramfs'es are huge (usually over 10 megs in size uncompressed)" is a little outdated. The generic initramfs is more around 100 MB and uncompressed in hundreds of megabytes.

pvalena commented:
LGTM, although I've not verified the package lists.
